lib/components/result-view/index.js

```diff
@@ -0,0 +1,107 @@
+/* @flow */
+
+import { CompositeDisposable, Point } from "atom";
+import React from "react";
+import { action, observable } from "mobx";
+import { createImmutableOutput } from "@nteract/commutable";
+
+import { reactFactory } from "./../../utils";
+import ResultViewComponent from "./result-view";
+
+import type { IObservableArray, IObservableValue } from "mobx";
+
+import type { ImmutableOutput } from "@nteract/commutable/lib/types";
+
+import type { Output, StreamOutput } from "@nteract/commutable/lib/v4";
+
+const outputTypes = ["execute_result", "display_data", "stream", "error"];
+
+/**
+ * https://github.com/nteract/hydrogen/issues/466#issuecomment-274822937
+ * An output can be a stream of data that does not arrive at a single time. This
+ * function handles the different types of outputs and accumulates the data
+ * into a reduced output.
+ *
+ * @param {Array<Object>} outputs - Kernel output messages
+ * @param {Object} output - Outputted to be reduced into list of outputs
+ * @return {Array<Object>} updated-outputs - Outputs + Output
+ */
+export function reduceOutputs(
```

Comment:
For this particular reducer we cannot use the one from nteract since it uses a immutable list which probably won't work well with MobX.

That said I really like the idea of exporting more nteract internals.